src/scenic/core/regions.py

```diff
@@ -2720,6 +2937,10 @@ def footprint(self):
     def containsPoint(self, point):
         return self.footprint.containsPoint(point)
 
+    @distributionFunction
+    def _trueContainsPoint(self, point):
+        return point.z == self.z and self.containsPoint
```

Comment:
```suggestion
        return point.z == self.z and self.containsPoint(point)
```
Better add a test which would have caught this!

Reply:
Good catch, fixed and added test!